### PR TITLE
spread: increase timeouts and fixes

### PR DESCRIPTION
**Pull request by mvo5**
There are currently a bunch of spread failures. It looks like they
might be related to the fact that the test take a long time. This
is also evident from seeing some "green" runs. This commit increases
the timeouts to have more consistent results.

### CODE REVIEW

**Review by mardy — APPROVED** [review]
Yes please :-)

**Comment by mvo5** [comment]
This worked once now, I trigger another build to see if it was just chance or if it really helps with the issue.

**Review by anonymouse64 — APPROVED** [review]
lgtm, sadly

**Comment by stulluk** [comment]
If I am not mistaken, this still fails in my "very slow" local system...

https://pastebin.ubuntu.com/p/4ttwPHT7Fz/

File: spread.yaml
Comment (anonymouse64) [inline spread.yaml:89]: maybe try 70m as the kill-timeout and 50m as the warn-timeout?

**Comment by mvo5** [comment]
Fwiw, after adding the debug output it appears that the swtpm socket was not ready. So qemu was briefly in "active" state and then failed a couple of seocnds later but the code in "wait_ssh" is not smart enough to notice that nested-vm-* is no longer active. I pushed fixes for the above issues. Maybe there is more though, let's see.

File: tests/lib/nested.sh
Comment (anonymouse64) [inline tests/lib/nested.sh:132]: as terrible and snapd specific as the nested.sh file in snapd is, it would probably serve us well to make that sharable with this repo, since the fixes you have here are also implemented in nested.sh in snapd now, and we have separate ways of handling this error, and I'm sure there are more errors.

but anyways that is for another time I suppose, this fix looks good to me

File: tests/lib/nested.sh
Comment (anonymouse64) [inline tests/lib/nested.sh:35]: will this race with the previous command `systemctl start`? I.e. maybe the ActiveState will be "activating" or something?

**Review by anonymouse64 — APPROVED** [review]
lgtm, good catch with the socket. 

As I mention below, it would be cool if we figured out how to share code between snapd and this repo for i.e. nested.sh

Also a question about a new possible race between systemctl start and systemctl show given the various states that ActiveState can have (i.e. `activating`). I don't think there's a race but I am always forgetting all of the systemd interactions about what state things can be in at what times after which commands

File: tests/lib/nested.sh
Comment (mvo5) [inline tests/lib/nested.sh:134]: Sorry, I'm a Muppet - this loop of course should decrease retry and eventually stop instead of looping forever (spread will kill it eventually of course but still a bit silly). :+100: on the idea to share code from snapd, just the fact that there is no "retry" helper is a major hindrance. Oh well - one step at a time :)

Comment (anonymouse64) [reply tests/lib/nested.sh:134]: Apparently /me is more of a Muppet for not noticing /o\

**Comment by mvo5** [comment]
One more successful run, will restart again.

**Comment by mvo5** [comment]
Second successful run, one more and I call the issue resolved :) (still needs addressing of the points that were raised in the code review.

**Comment by stulluk** [comment]
@mvo5 thank you very much. I confirm it finally works in my low spec machine :

https://pastebin.ubuntu.com/p/6dWn7sGfQW/

Note: testing takes about 2 hours in my system.

Approving.

**Review by stulluk — APPROVED** [review]

**Review by alfonsosanchezbeato — APPROVED** [review]
LGTM, thanks!

**Comment by mvo5** [comment]
Thanks, marking blocked until I manage to address the feedback (hopefully quick) - THANKS for all the reviews.

File: tests/lib/nested.sh
Comment (xnox) [reply tests/lib/nested.sh:35]: `systemctl start` is synchronous by default and waits for the unit to start. The asynchronous version that doesn't wait requires `--no-block` flag which is not use there. So this shouldn't be racy.  Unless of course the svc-script is racy and is not executing/started when it first spawns things. So this looks ok to me.

File: tests/lib/nested.sh
Comment (mvo5) [reply tests/lib/nested.sh:132]: Yeah, I asked Sergio to think about how we can share code between projects. I think you are spot-on here!

File: tests/lib/nested.sh
Comment (mvo5) [reply tests/lib/nested.sh:139]: I fixed this now

**Comment by mvo5** [comment]
This is ready now (I addressed the issues from the review) and this can be merged when it's green.

**Review by mardy — APPROVED** [review]

**Comment by xnox** [comment]
Note I had to retry this once to pass. Not sure if others were clicking retry as well or not. But hopefully things should get better with this in place.

**Comment by mvo5** [comment]
Thanks! Let's keep an eye on it and see if there is any flakiness left.

**Pull request merged**
